include/villas/fpga/card.hpp

```diff
@@ -93,11 +93,11 @@ class PCIeCard : public Card {
 	lookupIp(const ip::IpIdentifier &id) const;
 
 	bool
-	mapMemoryBlock(const MemoryBlock &block);
+	mapMemoryBlock(std::shared_ptr<MemoryBlock> block);
```

Comment:
Passing a shared pointer here makes things slow :( Does the block really needs to be mutated? Why cant we pass a const reference here?

Reply:
avoids memory leaks